CMakeLists.txt

```diff
@@ -0,0 +1,154 @@
+
+add_subdirectory(${CLP_CORE_ROOT}/src/clp/string_utils)
+
+# Source files of clp-ffi-core
+# Note: we don't add headers hear since they are not considered to be a part of the project.
+set(SOURCE_FILES_CLP_FFI_CORE
+    ${CLP_CORE_ROOT}/src/clp/BufferReader.cpp
+    ${CLP_CORE_ROOT}/src/clp/ffi/ir_stream/decoding_methods.cpp
+    ${CLP_CORE_ROOT}/src/clp/ffi/ir_stream/encoding_methods.cpp
+    ${CLP_CORE_ROOT}/src/clp/ffi/ir_stream/utils.cpp
+    ${CLP_CORE_ROOT}/src/clp/ffi/encoding_methods.cpp
+    ${CLP_CORE_ROOT}/src/clp/ir/parsing.cpp
+    ${CLP_CORE_ROOT}/src/clp/ReaderInterface.cpp
+)
+
+# Header files and source files of clp-ffi-py
+# Note: headers should be included to allow CLion loads the project properly.
+set(SOURCE_FILES_CLP_FFI_PY_LIB_IR
+    ${CLP_FFI_PY_SRC_ROOT}/error_messages.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ExceptionFFI.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/deserialization_methods.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/deserialization_methods.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/LogEvent.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/Metadata.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/Metadata.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyDeserializer.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyDeserializer.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyDeserializerBuffer.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyDeserializerBuffer.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyFourByteSerializer.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyFourByteSerializer.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyLogEvent.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyLogEvent.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyMetadata.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyMetadata.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyQuery.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/PyQuery.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/Query.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/Query.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/serialization_methods.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/ir/native/serialization_methods.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/modules/ir_native.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/Py_utils.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/Py_utils.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/PyObjectCast.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/PyObjectUtils.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/Python.hpp
+    ${CLP_FFI_PY_SRC_ROOT}/utils.cpp
+    ${CLP_FFI_PY_SRC_ROOT}/utils.hpp
+)
+
+target_sources(
+    ${CLP_FFI_PY_LIB_IR}
+    PRIVATE
+        ${SOURCE_FILES_CLP_FFI_CORE}
+        ${SOURCE_FILES_CLP_FFI_PY_LIB_IR}
+)
+
+target_include_directories(
+    ${CLP_FFI_PY_LIB_IR}
+    SYSTEM
+    PRIVATE
+        ${CLP_CORE_ROOT}/src
+        ${CLP_CORE_ROOT}/submodules
+        ${SRC_ROOT}
+)
+
+target_link_libraries(${CLP_FFI_PY_LIB_IR} PRIVATE clp::string_utils)
+
+if(CLP_FFI_PY_INSTALL_LIBS)
+    install(
+        TARGETS
+            ${CLP_FFI_PY_LIB_IR}
+            DESTINATION
+            ${CLP_FFI_PY}/ir
+            COMPONENT
+            PythonModule

```

Comment:
Sorry it's from this example and I directly copied the install command: https://github.com/scikit-build/scikit-build-core/blob/main/tests/packages/simplest_c/CMakeLists.txt
Seems like it might not be necessary